app/scripts/lib/channels/notifier-mixin.js

```diff
@@ -135,28 +163,31 @@ define(function (require, exports, module) {
     off (eventName, callback) {
       if (! eventName) {
         // unregister all callbacks for consumer.
-        this._notifierMessages.forEach(function (envelope) {
-          this._notifier.off(envelope.name, envelope.callback);
-        }, this);
+        this._notifierMessages.forEach(({ callback, eventName }) => {
+          this._notifier.off(eventName, callback);
+        });
         this._notifierMessages = [];
         return;
       }
 
-      this._notifierMessages.forEach(function (envelope, index) {
-        if (envelope.name === eventName && envelope.callback === callback) {
-          this._notifierMessages.splice(index, 1);
-          this._notifier.off(envelope.name, envelope.callback);
-        }
-      }, this);
+      // To simplify the code, the notification is not removed
+      // from this._notifierMessages. _notifierMessages is cleared
+      // when the object is destroyed, there is no negative
+      // side effect to attempting to remove a non-existent handler
+      this._notifier.off(eventName, callback);
     }
   };
 
   var NotifierMixin = {
-    initialize (options) {
-      this.notifier = new NotifierProxy({
-        consumer: this,
-        notifier: options.notifier
-      });
+    initialize (options = {}) {
+      // if no notifier is passed in, don't bother setting up
+      // the mixin. This avoids breaking all kinds of unit tests.
+      if (options.notifier) {
```

Comment:
Why did I never think of doing this all the times I've introduced a new option somewhere? Slavishly updating unit tests for this stuff is THE WORST.